### PR TITLE
Add the possibility to have two files in entry and then merge their info

### PR DESCRIPTION
**Pull request by calixteman**

### CODE REVIEW

File: src/linux/elf.rs [hunk 1]
```diff
@@ -397,6 +353,72 @@ impl ElfInfo {
             stack,
         })
     }
+
+    pub fn merge(left: ElfInfo, right: ElfInfo) -> ElfInfo {
```
Comment (gabrielesvelto) [inline src/linux/elf.rs:357]: I must admit I'm a bit stumped by this. I thought that symbolic would already have functionality to present a single view of a PE+PDB or .so+.dbg couple. It's a bit sad to see we have to merge the debuginfo ourselves.

File: src/linux/elf.rs [hunk 2]
```diff
@@ -397,6 +353,72 @@ impl ElfInfo {
...
+        // Just to avoid to iterate on the bigger
+        let (mut left, mut right) = if left.symbols.len() > right.symbols.len() {
+            (left, right)
+        } else {
+            (right, left)
+        };
```
Comment (gabrielesvelto) [inline src/linux/elf.rs:363]: I assume the output will be the same regardless of the order of the two symbol tables. It would be nice to add a test to ensure that's the case though maybe it would be best to do that with actual data instead. I was thinking that besides the unit-test we might want to setup a small test suite where we run this against a bunch of known debuginfos and match the output. Anyway this is not something to block on now.

File: src/linux/elf.rs [hunk 3]
```diff
@@ -178,21 +163,6 @@ impl InlineeManager {
...
-        if let Some(inlinee_pos) = self.addresses.get(&line.address) {
-            let inlinee = &mut self.inlinees[*inlinee_pos];
-            if inlinee.start == line.address {
-                let file_id = source.get_id(compilation_dir, &line.file);
-                inlinee.location = ElfLineInfo {
-                    file_id,
-                    line: line.line as u32,
-                };
-            }
-        }
```
Comment (gabrielesvelto) [inline src/linux/elf.rs:194]: This and the other changes to the inlined functions seem unrelated to the multi-file support. Is this the functionality that was recently merged into symbolic? If it is it would be best to split it out into another patch for clarity.

File: src/action.rs [hunk 4]
```diff
@@ -48,51 +49,85 @@ impl Dumper<'_> {
...
     }
 
-    fn guess(&self, path: PathBuf, filename: String) -> common::Result<()> {
-        let (buf, filename) = self.get_from_id(&path, filename, self.code_id)?;
+    fn two_elfs(
+        &self,
+        buf_1: Vec<u8>,
+        filename_1: String,
+        buf_2: Vec<u8>,
+        filename_2: String,
```
[the remark this answers is not in the document]
Comment (calixteman) [reply src/action.rs:101]: Agreed, we can do that in an other commit

File: src/linux/elf.rs (hunk 3 above)
Comment (calixteman) [reply src/linux/elf.rs:194]: I moved that stuff in a separate commit.
